content-test/faker.js

```diff
@@ -0,0 +1,179 @@
+const faker = require("faker");
+const moment = require("moment");
+const tiptop = require("tippy-top-sites");
+
+const BASE_TIP_TOP_FAVICON_URL = "favicons/images/";
+
+faker.timestamp = () => {
+  return moment().unix();
+};
+
+faker.internet.rgbColor = () => {
+  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(faker.internet.color());
+  return [
+    parseInt(result[1], 16),
+    parseInt(result[2], 16),
+    parseInt(result[3], 16)
+  ];
+};
+
+function arrayOf(mapFn, size) {
+  return Array.apply(null, Array(size)).map(mapFn);
+}
+
+function createColor(color) {
+  return {
+    "color": color || faker.internet.rgbColor(),
+    "weight": 0.3051757812
+  };
+}
+
+function createImage() {
+  return {
+    "caption": null,
+    "url": faker.image.imageUrl() + "?r=" + faker.random.uuid(),
+    "height": 640,
+    "width": 480,
+    "colors": [createColor()],
+    "entropy": 1.2698034041000001
+  };
+}
+
+const CREATE_SITE_DEFAULTS = {
+  images: 1,
+  override: null,
+  type: "history",
+  hasBookmark: false
+};
+
+function createSite(options = {}) {
+  options = Object.assign({}, CREATE_SITE_DEFAULTS, options);
+
+  const date = options.moment || moment().subtract(10, "seconds");
+
+  // Create the basic structure
+  const name = faker.company.companyName();
+
+  // TODO: make this optional
+  const tiptopSite = faker.random.arrayElement(tiptop);
+
+  const site = {
+    // This is stuff from Firefox
+    "bookmarkDateCreated": null,
+    "bookmarkGuid": null,
+    "favicon": "data:image/png;base64,iVBORw0KGgoAA",
+    "title": tiptopSite.title || `${name} - ${faker.company.catchPhrase()}`,
+    "url": tiptopSite.url || faker.internet.url(),
+    "lastVisitDate": date.valueOf(),
+    "frecency": faker.random.number({min: 10, max: 2000}),
+    "type": options.type,
+
+    // This is embedly
+    "description": faker.lorem.sentences(),
+    "favicon_url": BASE_TIP_TOP_FAVICON_URL + tiptopSite.image_url || faker.image.imageUrl(),
+    "media": {},
+    "provider_name": name
+  };
+
+  if (options.override) {
+    Object.assign(site, options.override);
+  }
+
+  // Add images if options.images > 0
+  if (options.images) {
+    site.images = arrayOf(createImage, options.images);
+  }
+
+  // Only create favicon_colors if there is a favicon_url
+  if (site.favicon_url && tiptopSite.background_color_rgb) {
```

Comment:
@k88hudson: Great idea! What would you need/want for an API?

If it's just to dynamically inject the `rgb[]` color array, it may be as easy as adding a "server.js" with the following (and then tweaking the main entry point in package.json):

```js
var hexToRgb = require("hex-to-rgb");

var data = require("./top_sites.json");

module.exports = data.map(function (site) {
  site.background_color = site.background_color.toUpperCase();
  site.background_color_rgb = hexToRgb(site.background_color);
  return site;
});
```

Or if you want a base64 version of the `image_url` that should be really simple too (although will add a considerable amount of weight to the JSON payload).